# ros_control_boilerplate Settings -----------------------
# Settings for ros_control control loop

generic_hw_control_loop:
   loop_hz: 10 # 与机械臂的通信频率
   cycle_time_error_threshold: 0.1 # 通信周期误差阈值，当实际通信周期达不到指定值时，会显示 warning

# Settings for ros_control hardware interface (used in generic_hw_interface.cpp)
hardware_interface:
   joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - fake_joint6 # 假关节，用于使逆解有解

# Publish all joint states ----------------------------------
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
   type: joint_state_controller/JointStateController
   publish_rate: 10

# Joint Trajectory Controller -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
position_trajectory_controller:
   type: position_controllers/JointTrajectoryController
   # These joints can likely just be copied from the hardware_interface list above
   joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - fake_joint6

joint_group_position_controller:
   type: position_controllers/JointGroupPositionController
   # These joints can likely just be copied from the hardware_interface list above
   joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - fake_joint6
